sqlmesh/core/model/meta.py

```diff
@@ -238,6 +242,29 @@ def _depends_on_validator(cls, v: t.Any, values: t.Dict[str, t.Any]) -> t.Option
 
         return v
 
+    @validator("grain", pre=True)
+    def _grain_validator(cls, v: t.Any, values: t.Dict[str, t.Any]) -> t.Optional[Reference]:
+        if isinstance(v, (exp.Expression, str)):
+            return Reference(expression=v, unique=True, dialect=values.get("dialect"))
+        return Reference(**v)
+
+    @validator("grains", "references", pre=True)
+    def _refs_validator(
```

Comment:
Why do we need this whole overhead of storing `Reference` instances as part of the model itself? Seems to be an overkill to just set the `unique` flag, especially provided this information is useless in the model definition itself.

Why can't we construct `Reference` instances on a fly when we add a model to the `Graph`? This will keep the model definition clean.

Reply:
we use references in other places like table diff, i can change it to be a customer getter